provers/sp1/guest/src/main.rs

```diff
@@ -1,28 +1,43 @@
 #![no_main]
 sp1_zkvm::entrypoint!(main);
 
+mod zk_op;
+
 use raiko_lib::{
     builder::{BlockBuilderStrategy, TaikoStrategy},
     input::{GuestInput, GuestOutput, WrappedHeader},
+    protocol_instance::{assemble_protocol_instance, EvidenceType},
 };
-use raiko_lib::protocol_instance::assemble_protocol_instance;
-use raiko_lib::protocol_instance::EvidenceType;
+use revm_precompile::zk_op::Operation;
+use zk_op::Sp1Operator;
 
 pub fn main() {
-
     let input = sp1_zkvm::io::read::<GuestInput>();
+
+    revm_precompile::zk_op::ZKVM_OPERATOR.get_or_init(|| Box::new(Sp1Operator {}));
+    revm_precompile::zk_op::ZKVM_OPERATIONS
+        .set(Box::new(vec![
+            Operation::Bn128Add,
+            Operation::Bn128Mul,
+            Operation::Secp256k1,
+        ]))
+        .expect("Failed to set ZkvmOperations");
```

Comment:
Do we still need this? Because Either trait can return an error for a specific zk function or just call the standard rust code itself. Seems like now two mechanisms to enable/disable certain zk operations.

Reply:
there's only one mechanism, we inject the ZkOp functions if this ZkVm impl this optimized circuit, otherwise don't do anything